File: backend/billing/apple_iap.py
"""Future Apple In-App Purchase verification.

Not implemented. Do not grant Premium from a client-supplied receipt,
transaction id, or on-device purchase flag.

When this is implemented:

- Verify signed transactions with Apple on the server
- Store Apple original_transaction_id separately from Stripe customer/subscription ids
- Grant Premium only after Apple verification succeeds
- Keep ``user_has_premium`` as the single entitlement predicate
"""

APPLE_IAP_PREMIUM_MONTHLY_PRODUCT_ID = "com.jduclos.flowsight.premium.monthly"
APPLE_IAP_VERIFY_PATH = "/api/billing/apple/verify/"


def apple_iap_verification_enabled() -> bool:
    return False

File: packages/shared/src/appleIap.ts
/**
 * Future Apple In-App Purchase types.
 *
 * Purchases are not implemented. Clients must not invent receipts or grant
 * Premium from device-only state. Server verification is required.
 */

/** Intended App Store product id. Create the product in App Store Connect before use. */
export const APPLE_IAP_PREMIUM_MONTHLY_PRODUCT_ID = "com.jduclos.flowsight.premium.monthly";

/** Planned backend verification path. Not routed yet. */
export const APPLE_IAP_VERIFY_PATH = "/api/billing/apple/verify/";

export type AppleIapEnvironment = "sandbox" | "production";

export type AppleIapPurchaseRequest = {
  productId: string;
};

export type AppleIapRestoreRequest = {
  productId?: string;
};

export type AppleIapStatus = {
  productId: string | null;
  originalTransactionId: string | null;
  expiresAt: string | null;
  environment: AppleIapEnvironment | null;
  isActive: boolean;
};

export type AppleIapVerificationPayload = {
  signedTransaction: string;
  productId: string;
};

export type AppleIapClient = {
  purchasePremium: (request: AppleIapPurchaseRequest) => Promise<AppleIapStatus>;
  restorePurchases: (request?: AppleIapRestoreRequest) => Promise<AppleIapStatus>;
  getPurchaseStatus: () => Promise<AppleIapStatus>;
};
